Start some docs for the operator (#2076)
kroxylicious/kroxylicious: docs/modules/configuring/con-configuring-kafkaservice-cipher.adoc

[id='con-configuring-kafkaservice-cipher-{context}']

> = Controlling the TLS ciphers used

Reviewer: ```suggestion
= Controlling TLS cipher suites
```

> Some cipher suites, which are enabled by default for interoperability purposes, are no longer consider secure.

Reviewer: Perhaps this needs more context?

_Some older cipher suites enabled by default for TLS connections between clients and virtual clusters are now considered insecure._

> IMPORTANT: It is considered good practice to disable insecure cipher suites. You can only do this is the proxied Kafka cluster you want to connect to supports the cipher suites that you want to use.

Reviewer: ```suggestion
IMPORTANT: It is good practice to disable insecure cipher suites. Only do this if the proxied Kafka cluster supports the cipher suites you want to allow.
```

Reviewer: Should this be a note or in the main body to introduce what we're showing in the config?

> The cipher suites that are enabled by default can depend on the JVM used in the proxy image and the TLS protocol version which is negotiated.

Reviewer: ```suggestion
The cipher suites enabled by default depend on the JVM used in the proxy image and the TLS protocol version that is negotiated.
```

Reviewer: I would change the order and bring this together with the sentence on _some cipher suites_

.Example `KafkaService` configured so that the proxy will negotiate TLS connection using only the listed ciphers.
[source,yaml]
----
kind: KafkaService
metadata:
  # ...
spec:
  bootstrapServers: kafka.example.com:9092
  tls:
    # ...
    cipherSuites: # <1>
      allowed: # <2>
       - TLS_AES_128_GCM_SHA256
       - TLS_AES_256_GCM_SHA384
----
<1> The `cipherSuites` object configures the cipher suites.
<2> `allowed` lists the cipher suites which are permitted.

> The `cipherSuites` object also supports `denied` if you prefer to exclude the unpermitted cipher suites, rather than include the permitted cipher suites.

Reviewer: ```suggestion
The `cipherSuites` property also supports `denied`, if you prefer to list the cipher suites to exclude instead.
```

[id='con-configuring-kafkaservice-cipher-{context}']
= Controlling TLS cipher suites

> A cipher suite is a set of cryptographic algorithms that together provide the security guarantees offered by TLS.

Reviewer: This intro reads much better now. :+1:
